ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/validator/SubmissionValidator.scala

```diff
@@ -75,15 +75,18 @@ class SubmissionValidator[LogResult] private[validator] (
       participantId: ParticipantId,
   ): Future[Either[ValidationFailed, LogResult]] =
     newLoggingContext { implicit loggingContext =>
-      validateAndCommitWithLoggingContext(envelope, correlationId, recordTime, participantId)
+      validateAndCommitWithContext(envelope, correlationId, recordTime, participantId)
     }
 
-  private[validator] def validateAndCommitWithLoggingContext(
+  private[validator] def validateAndCommitWithContext(
       envelope: Bytes,
       correlationId: String,
       recordTime: Timestamp,
       participantId: ParticipantId,
-  )(implicit loggingContext: LoggingContext): Future[Either[ValidationFailed, LogResult]] =
+  )(
+      implicit executionContext: ExecutionContext,
```

Comment:
I'd consider being a bit more audacious and:

- Removing the execution context from the constructor.
- Perhaps even removing `validate` that seems to be used in tests only (and possibly replacing it with a local or shared test utility) or pass the execution context there too.

Reply:
I'd like to do that but didn't want to break compatibility. If you're happy for me to do so, I'm on it.

Reply:
Looking into this, this means a larger breaking change, adding `implicit executionContext` to all methods of `LedgerStateAccess` and `LedgerStateOperations`.

I think this is a good idea, personally, but it would break source compatibility for implementors, and so I don't want to do it here. I'm happy to look at it separately.

Reply:
Yes you're right, better to evaluate separately.